deal.II 9.7.0: dof_accessor.cc does not compile on Visual Studio 2022 with C++20

OS: Windows 11 23H2

deal.II version: 9.7.0

Compiler: Visual Studio 2022 17.14.0

CMake commands:

`cmake -G "Visual Studio 17 2022" -A x64 -T host=x64 -DCMAKE_INSTALL_PREFIX=C:/sw/dealii/msvc-2022/9.7.0 -DDEAL_II_CXX_FLAGS="/std:c++20" -DDEAL_II_CXX_FLAGS_RELEASE="/MD" -DDEAL_II_DEFINITIONS="BOOST_ALL_DYN_LINK;BOOST_ALL_NO_LIB" -DDEAL_II_ALLOW_AUTODETECTION=OFF -DDEAL_II_WITH_COMPLEX_VALUES=ON -DDEAL_II_FORCE_BUNDLED_KOKKOS=ON -DDEAL_II_FORCE_BUNDLED_TASKFLOW=ON -DBOOST_DETECTED_TOOLSET=vc143 -DBoost_USE_STATIC_LIBS=OFF -DBoost_USE_STATIC_RUNTIME=OFF -DBOOST_DIR=%BOOST_ROOT% -S . -B build`

`cmake --build build --config DebugRelease`

I get the following compilation error:

```
1>dof_accessor.cc
1>C:\sw-dev\src\DealII\dealii-9.7.0\include\deal.II\dofs\dof_accessor.h(1294,15): error C7602: 'dealii::DoFHandler': the associated constraints are not satisfied
1>(compiling source file '../../../source/dofs/dof_accessor.cc')
1>    C:\sw-dev\src\DealII\dealii-9.7.0\include\deal.II\dofs\dof_handler.h(324,7):
1>    see declaration of 'dealii::DoFHandler'
1>    C:\sw-dev\src\DealII\dealii-9.7.0\include\deal.II\grid\tria_accessor.h(64,1):
1>    the concept 'dealii::concepts::is_valid_dim_spacedim<2,1>' evaluated to false
1>        C:\sw-dev\src\DealII\dealii-9.7.0\include\deal.II\base\template_constraints.h(728,39):
1>        the constraint was not satisfied
1>    C:\sw-dev\src\DealII\dealii-9.7.0\include\deal.II\dofs\dof_accessor.h(1294,15):
1>    the template instantiation context (the oldest one first) is
1>        C:\sw-dev\src\DealII\dealii-9.7.0\build20\source\dofs\dof_accessor.inst(1596,6):
1>        see reference to class template instantiation 'dealii::DoFInvalidAccessor<1,2,1>' being compiled
```

I don't know why, but it does compile with GCC 14.3 and C++20...

--- a/source/dofs/dof_accessor.inst.in
+++ b/source/dofs/dof_accessor.inst.in
@@ -67,7 +67,9 @@ for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS;
 for (deal_II_struct_dimension : DIMENSIONS; deal_II_dimension : DIMENSIONS;
      deal_II_space_dimension : SPACE_DIMENSIONS)
   {
+#if deal_II_dimension <= deal_II_space_dimension
     template class DoFInvalidAccessor<deal_II_struct_dimension,
                                       deal_II_dimension,
                                       deal_II_space_dimension>;
+#endif
   }

